crux-core/src/crux/kv_indexer.clj

```diff
@@ -761,7 +777,11 @@
      :crux.tx-log/consumer-state (db/read-index-meta this :crux.tx-log/consumer-state)}))
 
 (def kv-indexer
-  {:start-fn (fn [{:crux.node/keys [kv-store]} args]
+  {:start-fn (fn [{:crux.node/keys [kv-store]} {::keys [z-curve-index?]}]
                (check-and-store-index-version kv-store)
-               (->KvIndexer kv-store))
-   :deps [:crux.node/kv-store]})
+               (check-z-curve-index-config kv-store z-curve-index?)
+               (->KvIndexer kv-store z-curve-index?))
+   :deps [:crux.node/kv-store]
+   :args {::z-curve-index? {:crux.config/type :boolean
+                            :default false
```

Comment:
I think the default should be true (that is, stay as is), as the cost can be quite high when it's not there for certain data sets.